### ISSUE TITLE
weight in responses

### ISSUE THREAD

**Issue by gerhard-gs68** [aichaos/rivescript-python#102]
Just starting to work through the tutorial, it seems that the weight in responses doesn't work as expected. While executing many times

```
+ greetings
- Hi there!{weight=20}
- Hello!{weight=25}
- Yos kyoco duckeb!
```
responses come in not weighted. Cause is probably the implementation of weights (brain.py line 376)

 ```
                #Process weights in the replies.
                bucket = []
                for text in matched["reply"]:
                    weight = 1
                    match  = re.search(RE.weight, text) # original re.match
                    if match:
                        weight = int(match.grou
```
that is my modified version originally it used re.match instead of re.search, but match only matches at the beginnig of a string (if I recall correctly)

**Comment by Dinh-Hung-Tu** [comment]
@gerhard-gs68 You're correct at identifying this bug. You can make a merge request to fix this?

**Comment by gerhard-gs68** [comment]
@Dinh-Hung-Tu never done before on github but I will studyand give it a try with next week.

**Comment by Dinh-Hung-Tu** [comment]
@gerhard-gs68 Can I help you do this fix if you don't have time?

**Comment by kirsle** [comment]
Fixed by PR #106
